lib/msal-common/src/unifiedCache/UnifiedCacheManager.ts

```diff
@@ -26,33 +38,71 @@ export class UnifiedCacheManager {
     }
 
     /**
-     * setter  for in cache memory
+     * sets the inMemory cache
+     * @param cache
      */
     setCacheInMemory(cache: InMemoryCache): void {
         this.inMemoryCache = cache;
     }
 
     /**
-     * get the cache in memory
+     * get the inMemory Cache
      */
     getCacheInMemory(): InMemoryCache {
         return this.inMemoryCache;
     }
 
     /**
      * Initialize in memory cache from an exisiting cache vault
+     * @param cache
      */
     generateInMemoryCache(cache: string): InMemoryCache {
-        return Deserializer.deserializeAllCache(Deserializer.deserializeJSONBlob(cache));
+        return Deserializer.deserializeAllCache(
+            Deserializer.deserializeJSONBlob(cache)
+        );
     }
 
     /**
      * retrieves the final JSON
+     * @param inMemoryCache
      */
     generateJsonCache(inMemoryCache: InMemoryCache): JsonCache {
         return Serializer.serializeAllCache(inMemoryCache);
     }
 
+    /**
+     * append credential cache to in memory cach
+     * @param accessToken
+     * @param idToken
+     * @param refreshToken
+     */
+    addCredentialCache(
+        accessToken: AccessTokenEntity,
+        idToken: IdTokenEntity,
+        refreshToken: RefreshTokenEntity
+    ): void {
+        this.inMemoryCache.accessTokens[
+            accessToken.generateCredentialKey()
+        ] = accessToken;
```

Comment:
Can you add setter functions to `InMemoryCache`? This is leaking implementation details.

Reply:
Done